app/controllers/application_controller.rb

```diff
@@ -5,6 +5,144 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, 'totally_secure_key'
   end
 
-end
\ No newline at end of file
+  get "/" do
+    if logged_in?
+      redirect "/tweets"
+    else
+      erb :index
+    end
+  end
+
+  get "/signup" do
+    if logged_in?
+      redirect "/tweets"
+    else
+      erb :signup
+    end
+  end
+
+  post "/signup" do
+    user = User.new(params)
+    if user.save
+      session[:user_id] = user.id
+      redirect "/tweets"
+    else
+      redirect "/signup"
+    end
+  end
+
+  get "/login" do
+    if logged_in?
+      redirect "/tweets"
+    else
+      erb :login
+    end
+  end
+
+  post "/login" do
+    user = User.find_by(username: params[:username])
+    if user && user.authenticate(params[:password])
+      session[:user_id] = user.id
+      redirect "/tweets"
+    else
+      redirect "/login"
+    end
+  end
+
+  get "/logout" do
+    session.clear
+    redirect "/login"
+  end
+
+  get "/tweets" do
+    as_logged_in do
+      @tweets = Tweet.all
+      erb :'tweets/index'
+    end
+  end
+
+  post "/tweets" do
+    as_logged_in do |user|
+      tweet = Tweet.new(content: params[:content], user: user)
```

Comment:
Another way to do this is to make use of the `has_many` relationship on User:
```
tweet = current_user.tweets.create(content: params[:content])
```
Both approaches are good, but in some cases you may find this option useful.